apps/simplex-chat/Simplex/Chat.hs

```diff
@@ -160,6 +170,68 @@ chatSubscriber = do
             unsetActive $ ActiveC c
           _ -> pure ()
       _ -> pure ()
+  where
+    saveContact userId activeConn connInfo = do
+      p <- liftEither . first (ChatErrorContact . CEProfile) $ J.eitherDecodeStrict' connInfo
+      withStore $ \st -> createDirectContact st userId activeConn p
+
+getCreateActiveUser :: SQLiteStore -> IO User
+getCreateActiveUser st = do
+  user <-
+    getUsers st >>= \case
+      [] -> newUser
+      users -> maybe (selectUser users) pure (find activeUser users)
+  putStrLn $ "Current user: " <> userStr user
+  pure user
+  where
+    newUser :: IO User
+    newUser = do
+      putStrLn
+        "No user profiles found, it will be created now.\n\
+        \Please choose your alias and your profile name.\n\
+        \They will be sent to your contacts when you connect.\n\
+        \They are only stored on your device and you can change them later."
+      loop
+      where
+        loop = do
+          contactRef <- getContactRef
+          displayName <- T.pack <$> getWithPrompt "profile name (optional)"
+          liftIO (runExceptT $ createUser st Profile {contactRef, displayName} True) >>= \case
+            Left SEDuplicateContactRef -> do
```

Comment:
how can any contact exist before creation of the first user?

Reply:
it's the logic to allow adding profiles later using the same function.